src/overlays/actors/ovl_En_Arrow/z_en_arrow.c

```diff
@@ -33,45 +34,718 @@ const ActorInit En_Arrow_InitVars = {
+            }
+
+            if ((this->unk_262 != 0) && (this->unk_264 != NULL)) {
+                this->unk_264->flags &= ~0x8000;
+                this->unk_264 = NULL;
+            }
+        } else {
+            this->unk_264 = NULL;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Arrow/func_8088ACE0.s")
+#endif
+
+void func_8088B630(EnArrow* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->arrow.skelAnime);
+
+    if (this->actor.wallBgId != BG_ACTOR_MAX) {
+        BgCheck2_UpdateActorAttachedToMesh(&globalCtx->colCtx, this->actor.wallBgId, &this->actor);
+    }
+
+    if (DECR(this->unk_260) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_8088B6B0(EnArrow* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->arrow.skelAnime);
+    Actor_MoveWithGravity(&this->actor);
+    func_8088AA98(this, globalCtx);
+
+    if (DECR(this->unk_260) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnArrow_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static Vec3f D_8088C2CC = { 0.0f, 0.5f, 0.0f };
+    static Vec3f D_8088C2D8 = { 0.0f, 0.5f, 0.0f };
+    static Color_RGBA8 D_8088C2E4 = { 255, 255, 100, 255 };
+    static Color_RGBA8 D_8088C2E8 = { 255, 50, 0, 0 };
+    s32 pad;
+    EnArrow* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if ((this->unk_263 != 0) ||
+        ((this->actor.params >= ENARROW_0) && ((this->actor.params == ENARROW_7) || (player->unk_D57 != 0))) ||
+        !func_80123358(globalCtx, player)) {
+        this->actionFunc(this, globalCtx);
+    }
+
+    if ((this->actor.params >= ENARROW_3) && (this->actor.params < ENARROW_6)) {
+        s16 sp44[] = { ACTOR_ARROW_FIRE, ACTOR_ARROW_ICE, ACTOR_ARROW_LIGHT };
+
+        if (this->actor.child == NULL) {
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, sp44[this->actor.params - ENARROW_3],
+                               this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
+        }
+    } else if (this->actor.params == ENARROW_0) {
+        func_800B0EB0(globalCtx, &this->unk_234, &D_8088C2CC, &D_8088C2D8, &D_8088C2E4, &D_8088C2E8, 100, 0, 8);
+    }
+    Math_Vec3f_Copy(&this->actor.home.pos, &this->actor.prevPos);
+}
+
+typedef struct {
+    /* 0x00 */ Vec3f unk_00[2];
+    /* 0x18 */ Vec3f unk_18[2][2];
+    /* 0x48 */ Vec3f unk_48;
+} EnArrowUnkStruct; // size = 0x54
+
+void func_8088B88C(GlobalContext* globalCtx, EnArrow* this, EnArrowUnkStruct* arg2) {
+    Vec3f* sp4C;
+    Vec3f sp40;
+    Vec3f sp34;
+    s32 sp30;
+
+    Matrix_MultiplyVector3fByState(&arg2->unk_48, &this->unk_234);
+    if (func_8088ACE0 == this->actionFunc) {
+        if (this->unk_244 == 0) {
+            sp4C = &arg2->unk_00[0];
+        } else {
+            sp4C = &arg2->unk_18[globalCtx->gameplayFrames % 2][0];
```

Comment:
```suggestion
        if (this->unk_244 == 0) {
            sp4C = arg2->unk_00;
        } else {
            sp4C = arg2->unk_18[globalCtx->gameplayFrames % 2];
```
Minor, but this makes more sense to me since sp4C is suppose to be the full array, not just a pointer to the first element.